### PR TITLE
feat: slice login page

### PR DESCRIPTION
**Pull request by jeeehaan**
Preview:
image

### CODE REVIEW

File: src/components/(auth)/auth/register.tsx
Comment (fatjan) [inline src/components/(auth)/auth/register.tsx:1]: Can we please not name any directory/folder with () ?

**Review by fatjan — CHANGES_REQUESTED** [review]
Kindly update your directory naming as well as the import

File: src/components/(auth)/auth/register.tsx
Comment (jeeehaan) [reply src/components/(auth)/auth/register.tsx:1]: yes, we can. will do

Comment (jeeehaan) [reply src/components/(auth)/auth/register.tsx:1]: but in the `app` dir, i have to keep the parenthesis to group related routes. i'll just remove the parenthesis in components folder.

**Comment by jeeehaan** [comment]
updated @fatjan

**Review by fatjan — CHANGES_REQUESTED** [review]
You still have (auth) dir here. Kindly update it. Thanks.
Is it just me or?

Screenshot 2024-01-07 at 09 46 51

**Comment by jeeehaan** [comment]
I think i have to keep the parenthesis in the `app` dir to group related routes for the purpose to organize folder without affecting URL path. In this case, login and register can be grouped in `auth`, but i want the url be `/login` instead `/auth/login`.

reference:
https://nextjs.org/docs/app/building-your-application/routing/route-groups

**Review by fatjan — APPROVED** [review]

File: src/components/app/auth/login.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { Input } from '@/components/ui/input';
+
+export const Login = () => {
+  return (
+    <main className="flex h-screen items-center justify-center">
+      <div className="flex w-[300px] flex-col space-y-3">
+        <h3>Login</h3>
+        <p>Login to Indonesia Open Course</p>
+        <Input type="email" placeholder="Enter your email" />
+        <Input type="password" placeholder="Enter your password" />
```
Comment (ahmadrasyidsalim) [inline src/components/app/auth/login.tsx:13]: Hi, shouldn't we put `<input>` inside `<form>` attribute?

Comment (jeeehaan) [reply src/components/app/auth/login.tsx:13]: I don't usually use the `<form>` tag.

I usually code it like this

```tsx
  const [loginData, setLoginData] = useState({
    email: "",
    password: "",
  });
```

then a  handleEventChange function
```tsx
  const handleEventChange = (event) => {
    const { name, value } = event.target;
    setLoginData({ ...loginData, [name]: value });
  };
```

then a handleSubmit function
```tsx
 const handleSubmitLogin = async () => {
  const { email, password } = loginData;
  const res = await fetch(`${API_URL}/login`, {
    method: "POST",
    body: JSON.stringify({ email, password }),
    });
  const { data } = await res.json();
 };
```

Finally
```tsx
   <label>Email</label>
   <input name="email" onChange={handleEventChange} />
  <label>Password</label>
  <input name="password" onChange={handleEventChange} />
  <button onClick={handleSubmitLogin}>Login</button>
```

Comment (jeeehaan) [reply src/components/app/auth/login.tsx:13]: But should we put `<form>` tag in this case? @ahmadrasyidsalim

**Pull request merged**